aws/resource_aws_fsx_file_system_test.go

```diff
@@ -0,0 +1,365 @@
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: resourceName,
+		Providers:     testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsFsxFileSystemWindowsConfigOpts,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFileSystemExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.0.backup_retention", "3"),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.0.copy_tags_to_backups", "true"),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.0.throughput_capacity", "1024"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"kms_key_id"},
+			},
+		},
+	})
+}
+
+func TestAccAWSFsxFileSystem_windowsUpdate(t *testing.T) {
+	var v fsx.FileSystem
+	resourceName := "aws_fsx_file_system.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: resourceName,
+		Providers:     testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsFsxFileSystemWindowsConfigOpts,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFileSystemExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.0.backup_retention", "3"),
+				),
+			},
+			{
+				Config: testAccAwsFsxFileSystemWindowsUpdateOpts,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFileSystemExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.0.backup_retention", "30"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckFileSystemExists(n string, v *fsx.FileSystem) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).fsxconn
+
+		request := &fsx.DescribeFileSystemsInput{
+			FileSystemIds: []*string{aws.String(rs.Primary.ID)},
+		}
+
+		response, err := conn.DescribeFileSystems(request)
+		if err == nil {
+			if response.FileSystems != nil && len(response.FileSystems) > 0 {
+				*v = *response.FileSystems[0]
+				return nil
+			}
+		}
+		return fmt.Errorf("Error finding FSx filesystem %s", rs.Primary.ID)
+	}
+}
+
+const testAccAwsFsxFileSystemLustreBasic = `
+resource "aws_vpc" "test" {
+  cidr_block  = "10.0.0.0/16"
+}
+
+resource "aws_subnet" "test" {
+  vpc_id     				= "${aws_vpc.test.id}"
+	cidr_block 				= "10.0.1.0/24"
+	availability_zone = "us-east-1a"
```

Comment:
As noted in the [Contributing Guide](https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/CONTRIBUTING.md#acceptance-testing-guidelines), acceptance testing should pass without modification in `us-west-2`. The easiest way to setup region agnostic testing is to use the `aws_availability_zones` data source, e.g.

```hcl
data "aws_availability_zones" "available" {
  state = "available"
}

resource "aws_subnet" "test" {
  availability_zone = "${data.aws_availability_zones.available.names[0]}"
  # ...
```

Reply:
Yeah, I'm not sure why this is, but I'm seeing my FSx resources kick off before the VPC/subnet are ready which is causing failures.  `depends_on` seems to be needed here for my tests to work but I agree that shouldn't be needed.